cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -184,15 +185,31 @@ public void removeRequest(ProgramRunId programRunId, boolean emitRunningChange)
     }
 
     if (emitRunningChange) {
-      emitMetrics(Constants.Metrics.FlowControl.RUNNING_COUNT, getProgramsRunningCount());
+      emitRunningMetrics();
     }
   }
 
   public void emitLaunchingMetrics(long value) {
     emitMetrics(Constants.Metrics.FlowControl.LAUNCHING_COUNT, value);
   }
 
+  /**
+   * Emit the {@link Constants.Metrics.FlowControl#LAUNCHING_COUNT} metric for runs.
+   */
+  public void emitLaunchingMetrics() {
+    emitMetrics(Constants.Metrics.FlowControl.LAUNCHING_COUNT, launchingQueue.size());
+  }
+
+
+  /**
+   * Emit the {@link Constants.Metrics.FlowControl#RUNNING_COUNT} metric for runs.
+   */
+  public void emitRunningMetrics() {
+    emitMetrics(FlowControl.RUNNING_COUNT, getProgramsRunningCount());
+  }
+
   private void emitMetrics(String metricName, long value) {
+    LOG.debug("Setting metric {} to value {}",metricName,value);
```

Comment:
checkstyle: add spaces after comma: `LOG.debug("Setting metric {} to value {}", metricName, value);`